controllers/component_controller_conditions.go

```diff
@@ -49,22 +51,21 @@ func (r *ComponentReconciler) SetCreateConditionAndUpdateCR(ctx context.Context,
 		logutil.LogAPIResourceChangeEvent(log, component.Name, "Component", logutil.ResourceCreate, createError)
 	}
 	meta.SetStatusCondition(&component.Status.Conditions, condition)
-
 	err := r.Client.Status().Update(ctx, component)
 	if err != nil {
 		// Retry, and if still fails, then return an error
-		curComponent := &appstudiov1alpha1.Component{}
-		err := r.Client.Get(ctx, req.NamespacedName, curComponent)
+		var currentComponent appstudiov1alpha1.Component
+		err := r.Get(ctx, req.NamespacedName, &currentComponent)
 		if err != nil {
-			return err
+			return nil
```

Comment:
didn't notice this before but did you mean to return nil instead of err?  If so can you update the comment on line 56 which says otherwise?

Reply:
updated